### DISCUSSION TITLE
Records in the table

### DISCUSSION THREAD

**Discussion by andim27** [simolus3/drift#2724] [General]
Flutter drift 2.11 How to count records in the table? 
Why I can't do : 
` int cnt = select(myTable).count();`

**Comment by simolus3** [comment]
> Flutter drift 2.11 How to count records in the table?

You need to add the aggregate expression explicitly:

```dart
final count = countAll();
final stmt = selectOnly(myTable)..addColumns([count]);
int cnt = await stmt.map((row) => row.read(count)!).getSingle();
```

> Why I can't do: ` int cnt = select(myTable).count();`

That's a fair question, I think we should have an API for this since it's fairly common. I'll add add extension allowing you to write `myTable.count().getSingle()` in the next version.

**Reply by bendangelo** [reply]
Has this been done?

**Reply by simolus3** [reply]
Yes, there's a [count](https://drift.simonbinder.eu/api/drift/tableorviewstatements/count) extension method on tables and views.
